Sources/LiveKit/Core/Room.swift

```diff
@@ -224,18 +224,24 @@ extension Room {
         }
 
         let sendUnSub = connectOptions?.autoSubscribe ?? false
-        let trackSids = remoteParticipants.values.map {
-            $0.tracks.values
-                .filter { $0.subscribed != sendUnSub }
-                .map { $0.sid }
-        }.flatMap { $0 }
+        let participantTracks = remoteParticipants.values.map { participant in
+            Livekit_ParticipantTracks.with {
+                $0.participantSid = participant.sid
+                $0.trackSids = participant.tracks.values
+                    .filter { $0.subscribed != sendUnSub }
+                    .map { $0.sid }
+            }
+        }
+
+        // Backward compatibility
+        let trackSids = participantTracks.map { $0.trackSids }.flatMap { $0 }
```

Comment:
Is this to send over trackSids for older server versions?

Reply:
Yup just in case, not sure if needed though 🙂

Reply:
yeah we should do this for a period of time. since upgrading the server is more painful, some of our users only upgrade every few months.

Reply:
Maybe we can gate this on protocol 6

Reply:
agreed. we'd just have to send it for pre-protocol6 server